src/api.ts

```diff
@@ -26,3 +26,35 @@ function get<T>(url: string): Promise<T> {
 export const getTodos = () => get<Todo[]>('/todos');
 
 export const getUser = (userId: number) => get<User>(`/users/${userId}`);
+
+export const getSortedTodos = () => {
+  return getTodos().then(todos => todos.sort((a, b) => a.id - b.id));
+};
+
+export const getCompletedTodos = () => {
+  return getSortedTodos().then(todos =>
+    todos.filter(todo => todo.completed === true),
+  );
+};
+
+export const getActiveTodos = () => {
+  return getSortedTodos().then(todos =>
+    todos.filter(todo => todo.completed === false),
+  );
+};
+
+export const getFilteredTodos = (value: string) => {
+  switch (value) {
+    case 'all':
+      return getSortedTodos();
+
+    case 'active':
+      return getActiveTodos();
+
+    case 'completed':
+      return getCompletedTodos();
+
+    default:
+      return getSortedTodos();
```

Comment:
The 'getFilteredTodos' function uses a string value to determine the filter type. It's a good practice to use TypeScript's enum or union types for such cases to prevent the risk of typos and to provide clearer documentation of the allowed values.